src/core/buffers/buffer_orchestrator.ts

```diff
@@ -247,47 +265,32 @@ export default function BufferOrchestrator(
 
     // Restart the current buffer when the wanted time is in another period
     // than the ones already considered
-    const restartBuffers$ = clock$.pipe(
+    const restartBuffersWhenOutOfBounds$ = clock$.pipe(
       filter(({ currentTime, wantedTimeOffset }) => {
-        return !!manifest.getPeriodForTime(wantedTimeOffset + currentTime) &&
+        return hasLoadedABuffer &&
+          !!manifest.getPeriodForTime(wantedTimeOffset + currentTime) &&
           isOutOfPeriodList(wantedTimeOffset + currentTime);
       }),
-
-      take(1),
-
-      tap(({ currentTime, wantedTimeOffset }) => {
+      mergeMap(({ currentTime, wantedTimeOffset }) => {
         log.info("Buffer: Current position out of the bounds of the active periods," +
           "re-creating buffers.", bufferType, currentTime + wantedTimeOffset);
-        destroyCurrentBuffers.next();
-      }),
-
-      mergeMap(({ currentTime, wantedTimeOffset }) => {
+        hasLoadedABuffer = false;
+        destroyBuffers$.next();
```

Comment:
This line could lead to a load of side-effect, might be more sure to isolate it in its own `tap` operator in case it provokes unsubscriptions and whatnot.